ARCH/openfpga_arch_template/k4_N8_reset_softadder_register_scan_chain_caravel_io_skywater130nm_fdhd_cc_openfpga.xml

```diff
@@ -309,19 +309,32 @@ foundry middle-speed (ms) standard cell library
       <port type="sram" prefix="sram" size="16"/>
     </circuit_model>
     <!-- new ccFF  -->
-    <circuit_model type="ccff" name="CFGSDFFR" prefix="CFGSDFFR" spice_netlist="${OPENFPGA_PATH}/openfpga_flow/openfpga_cell_library/spice/dff.sp" verilog_netlist="${OPENFPGA_PATH}/openfpga_flow/openfpga_cell_library/verilog/dff.v">
+    <circuit_model type="ccff" name="QL_CCFF" prefix="QL_CCFF" verilog_netlist="${SKYWATER_OPENFPGA_HOME}/HDL/common/ql_ccff.v">
       <design_technology type="cmos"/>
       <input_buffer exist="true" circuit_model_name="sky130_fd_sc_hd__inv_1"/>
       <output_buffer exist="true" circuit_model_name="sky130_fd_sc_hd__inv_1"/>
-      <port type="input" prefix="pReset" lib_name="RST" size="1" is_global="true" default_val="0" is_reset="true" is_prog="true"/>
+      <port type="input" prefix="pReset" lib_name="RESET_B" size="1" is_global="true" default_val="1" is_reset="true" is_prog="true"/>
       <port type="input" prefix="Test_en" lib_name="SE" size="1" is_global="true" default_val="0"/>
-      <port type="input" prefix="CFG_DONE" lib_name="CFGE" size="1" is_global="true" default_val="0" is_config_enable="true"/>
+      <port type="input" prefix="CFG_DONE" lib_name="CFGE" size="1" is_global="true" default_val="0"/>
       <port type="input" prefix="D" size="1"/>
       <port type="input" prefix="SI" size="1"/>
       <port type="output" prefix="Q" size="1"/>
       <port type="output" prefix="CFGQN" size="1"/>
       <port type="output" prefix="CFGQ" size="1"/>
-      <port type="clock" prefix="prog_clk" lib_name="CK" size="1" is_global="true" default_val="0" is_prog="true"/>
+      <port type="clock" prefix="prog_clk" lib_name="CLK" size="1" is_global="true" default_val="0" is_prog="true"/>
+    </circuit_model>
+    <!-- dummy stdcell pointer -->
+    <circuit_model type="inv_buf" name="dummy1" prefix="dummy1" verilog_netlist="${TASK_DIR}/skywater-pdk/libraries/sky130_fd_sc_hd/latest/cells/nor2/sky130_fd_sc_hd__nor2_1.v"> 
+      <design_technology type="cmos" topology="inverter" size="1"/> 
+    </circuit_model>
+    <circuit_model type="inv_buf" name="dummy2" prefix="dummy2" verilog_netlist="${TASK_DIR}/skywater-pdk/libraries/sky130_fd_sc_hd/latest/cells/nand2/sky130_fd_sc_hd__nand2_1.v"> 
+      <design_technology type="cmos" topology="inverter" size="1"/> 
+    </circuit_model>
+    <circuit_model type="inv_buf" name="dummy3" prefix="dummy3" verilog_netlist="${TASK_DIR}/skywater-pdk/libraries/sky130_fd_sc_hd/latest/cells/einvn/sky130_fd_sc_hd__einvn_4.v"> 
+      <design_technology type="cmos" topology="inverter" size="1"/> 
+    </circuit_model>
+    <circuit_model type="inv_buf" name="dummy4" prefix="dummy4" verilog_netlist="${TASK_DIR}/skywater-pdk/libraries/sky130_fd_sc_hd/latest/cells/and3/sky130_fd_sc_hd__and3_1.v"> 
+      <design_technology type="cmos" topology="inverter" size="1"/> 
```

Comment:
adding ${TASK_DIR} in order to include those stdcells into fabric_netlists.v for testbench

Reply:
Is ${TASK_DIR} an environment variable on your local machine? Alternatively, you can follow the same environment variable as used in other circuit models. There are python scripts converting the environment variable to absolute path, so that your architecture files are always updated to your local settings

Reply:
This was originally suggested by Ganesh.  After 2nd-thoughts, with explicit "dummy" circuit_model, there in no need to use speical env. variable.   Thanks. (unless, I am missing Ganesh's point)